### DISCUSSION TITLE
How can I add access log middleware?

### DISCUSSION THREAD

**Discussion by yardenshoham** [alephjs/aleph.js#566] [Q&A]
Like morgan

**Comment by ije** [comment]
you can add a middleware for that, for example:

```js

import { serve } from "aleph/server";

serve({
  /// ...your config,
  middlewares: [
   {
      name: "logger"
      fetch: async (req,ctx) => {
        await ctx.next()
        console.log(req.url)
      }
   }
  ],
});

```

**Reply by yardenshoham** [reply]
Can I use any existing libraries or do I have to implement the logging logic of getting the IP, user agent, status, response time, etc... myself?

Maybe https://github.com/expressjs/morgan or https://github.com/khuongduybui/fresh-logging?

**Reply by ije** [reply]
no sure, FYI the `req` argument is a standard `Request` object, and the `ctx.next()` is calling the next middleware.

**Reply by yardenshoham** [reply]
Is there a way to grab the response so I could log the response time and response status code?
